app/models/manageiq/providers/vmware/infra_manager/inventory/parser/virtual_machine.rb

```diff
@@ -302,7 +302,7 @@ def parse_virtual_machine_networks(vm, props, hardware, guest_devices)
         props.fetch_path(:guest, :net).to_a.each do |net|
           ip_config_by_ip_addr = net.ipConfig&.ipAddress&.index_by(&:ipAddress) || {}
 
-          ipv4, ipv6 = net[:ipAddress].to_a.compact.collect(&:to_s).sort.partition { |ip| ip =~ /([0-9]{1,3}\.){3}[0-9]{1,3}/ }
+          ipv4, ipv6 = net[:ipAddress].to_a.compact.collect(&:to_s).sort.partition { |ip| ip =~ /^([0-9]{1,3}\.){3}[0-9]{1,3}/ }
```

Comment:
```suggestion
          ipv4, ipv6 = net[:ipAddress].to_a.compact.collect(&:to_s).sort.partition(&:ipv4?)
```
We have a method to deal with this already in more_core_extensions...can we use that?

Reply:
```ruby
[1] pry(main)> require "more_core_extensions/all"
[2] pry(main)> "::ffff:127.0.1.1".ipv4?
=> false
[3] pry(main)> "::ffff:127.0.1.1".ipv6?
=> true
```

Reply:
Oh even better :+1:

Reply:
Since partition only checks one side of the equation, you _might_ want to do a second pass verification of the ipv6 ones?


```ruby
ipv4, rest = net[:ipAddress].to_a.compact.collect(&:to_s).sort.partition(&:ipv4?)
ipv6, rest = rest.partition(&:ipv6?)
```

Reply:
we could `select(&:ipaddress?).partition(&:ipv4)` to one-line it?